### ISSUE TITLE
 CompleteInput support to use the dir separator as the input on Windows #2219 

### ISSUE THREAD

**Issue by alanlivio** [PowerShell/PowerShell#19527]
### Summary of the new feature / enhancement

Some time ago a created this [issue in PSReadline](https://github.com/PowerShell/PSReadLine/issues/2219) about use tab-completion using a forward slash in PowerShell running on Windows (to simulate Unix better). For instance, today the `ls C:/Users` followed by the tab is automatically converted to `C:\Users\`. I would expect `C:/Users/`.

PSReadline team said it couldn't be done because [CompleteInput](https://learn.microsoft.com/en-us/dotnet/api/system.management.automation.commandcompletion.completeinput) is coded to use the system separator. 

Therefore, I feature request is that the CompleteInput return respects the same dir separator used in the input.

**Comment by ryan-beesley** [comment]
@alanlivio, I found this issue searching for a resolution to this exact scenario. I can think of reasons why this shouldn't be the default for all installations, but it'd make things easier for me if this was an option.

**Comment by alanlivio** [comment]
Hi @MartinGC94, any update on the PR? Thanks.

**Comment by MartinGC94** [comment]
The path settings have been implemented and if you want to try them out you can create a build from my branch. I'm actually just waiting for another PR of mine to get merged: https://github.com/PowerShell/PowerShell/pull/18955 so I can add those options as well.
It should be noted that PSReadLine interferes a little with my changes because of the way it auto inserts a separator for directories, see: https://github.com/PowerShell/PSReadLine/issues/3699 so that probably needs to be fixed as well before my changes makes its way into a production build.

**Comment by alanlivio** [comment]
Hi, again @MartinGC94. It seems that your [PR has been closed](https://github.com/PowerShell/PowerShell/pull/19518).

**Comment by MartinGC94** [comment]
Yeah I know. I'm still waiting on: https://github.com/PowerShell/PowerShell/pull/18955 to get merged. I could of course just leave that feature out for now and resubmit it as is but I suspect it won't make the PS team merge these changes any faster. It seems like they are not really merging any new community changes at the moment where they are preparing for an official 7.4.0 release.
My best estimate is that they'll "pick up the slack" in January and start merging old PRs and then hopefully it will make its way into a 7.5.0 preview 1 release.

**Comment by cyanite** [comment]
Recently, PowerShell, or PSReadLine, stopped expanding ~ to your user profile, which makes the slash issue much worse. Now, if I type a "less" command line (using git's less) like this:

    less ~/somefi

and press tab, it gets completed to

    less ~\somefile.txt

which doesn't work in less. Before, it completed to the full path (with \) which did work. Since I use less constantly, this is a pretty big annoyance.

**Comment by alanlivio** [comment]
Dear @cyanite, I am also very used to typing /.
The most important thing to me is that the windows will accept the / if you send it. But the PowerShell CompleteInput force used to use the \.

**Comment by cyanite** [comment]
> Dear @cyanite, I am also very used to typing /. The most important thing to me is that the windows will accept the / if you send it. But the PowerShell CompleteInput force used to use the .

Right. For me, the most important would be if I could tell it to always complete with /. But any variant will do :)

**Comment by dyskos** [comment]
I hope #18955 will make it in soon, the autocomplete forcing the use of backslash breaks my use of WSL commands from Windows.

I'll try to run commands like so:
`lx kak ./client.go # lx is an alias for wsl -e`
but the auto-complete changes it to:
`lx kak .\client.go`
which breaks how the command is interpreted by WSL and instead of opening the existing file, it opens a new empty text file in the WSL home directory.

Anytime I use a WSL command in Windows via PowerShell, I have to type the names/paths by hand to circumvent this problem, which is a nuisance to say the least.

**Comment by alanlivio** [comment]
Thanks @dyskos. I am really looking forward to that as well.

**Comment by Xiphoseer** [comment]
FWIW I came across this issue because [Jest](https://jestjs.io/) (javascript testing tool) seems to try to detect a path argument from a testcase name by checking whether it has forward slashes, which breaks in PowerShell.

**Comment by alanlivio** [comment]
For the ones looking for a solution until the @MartinGC94 MR be accepted, @mklement0 proposed a profile-based solution at https://github.com/PowerShell/PowerShell/discussions/16671#discussioncomment-1869254. I tried and it works.

<blockquote><img src="https://opengraph.githubassets.com/b8bcd4b015cc4ad0d96085b735d25f49ce868ee7560d735432a0816670afeb36/PowerShell/PowerShell/discussions/16671" width="48" align="right"><div><img src="https://github.githubassets.com/favicons/favicon.svg" height="14"> GitHub</div><div><strong><a href="https://github.com/PowerShell/PowerShell/discussions/16671">Use forward slashes by default in Windows · PowerShell PowerShell · Discussion #16671</a></strong></div><div>Summary of the new feature/enhancement PowerShell aims to be cross-platform, but I have been having many issues operating on paths across both Windows and Linux. I understand the need to support \ ...</div></blockquote>
